### ISSUE TITLE
Implement `kubernetes:kubernetes:list` via Invoke

### ISSUE THREAD

**Issue by hausdorff** [pulumi/pulumi-kubernetes#859]
The overall idea is to provide an experience like `kubectl get svc`, through the Kubernetes provider API—allowing users to enumerate all resources in the Kubernetes cluster of some type. For example, in principle, a user could write something like this:

```typescript
invoke("kubernetes:kubernetes:list", {
    group: "",
    version: "v1",
    kind: "Service",
})
```

Which would return, effectively, `Promise<{ items: any[] }>`, where `items` contains all the `Service` objects live in the Kubernetes cluster.

Part of this work is actually to expose a well-typed "query layer" on top of this, so that users can write something like:

```typescript
k8s.list("v1", "Service").forEach(o => {
    console.log(JSON.stringify(o));
});
```

Almost all of the core work in pulumi/pulumi will land in https://github.com/pulumi/pulumi/pull/3085; the PR that will solve most of this issue directly is https://github.com/pulumi/pulumi-kubernetes/pull/858.

**Comment by hausdorff** [comment]
This will be resolved by #858, but I wanted to have proposals for both `list` and whatever watch functionality we needed before committing to the API.

**Comment by hausdorff** [comment]
Resolved in `pulumi/pulumi-query-kuberntes`
